packages/gatsby/src/utils/babel-config.js

```diff
@@ -143,6 +143,11 @@ module.exports = async function babelConfig(program, stage) {
     babelrc.plugins.unshift(require.resolve(`react-hot-loader/babel`))
   }
 
+  // Make dynamic imports work during SSR.
+  if (stage === `build-html` || stage === `develop-html`) {
+    babelrc.plugins.unshift(require.resolve(`babel-plugin-dynamic-import-node`))
```

Comment:
This seems like you'd get different behavior between the different stages, I don't know that I have alternatives but how does require.ensure get handled for a similar case (e.g. In gatsby v1)

Reply:
require.ensure is fine I think since it's callback based so is ignored while SSRing. This plugin just makes Node not throw up by transpiling the dynamic import to a deferred require.

Reply:
so the issue is that node doesn't have the syntax support, not that the require does anything? Presumably if `require.ensure` was attempted it would break also since node doesn't have it yeah?

Reply:
Maybe... perhaps webpack compiles it to something else with the `umd` target.

Reply:
well i guess we haven't had any issues so for with it :P sorry just was curious about waht was happening

Reply:
:-D no worries. We also could disable it like CRA since it's a non-standard feature.